### PR TITLE
fix(ghe): update github release notes to use api endpoint instead of sourceUrl

### LINKED ISSUE #3079: Incorrect url used for getReleaseList call with Github Enterprise
**What Renovate type are you using?**
Renovate CLI

**Describe the bug**
When using renovate with Github Enterprise, the url used in `getReleaseList` is incorrect.

It appears to be using: 

`https://github.mycompany.com/repos/someowner/somepackage/releases?per_page=100`

When, instead, it should be using:

`https://github.mycompany.com/api/v3/repos/someowner/somepackage/releases?per_page=100`

We are configured with the following env variables (this used to work I think):
```
GITHUB_TOKEN=my-ghe-token
GITHUB_ENDPOINT=https://github.myoffice.com/api/v3/
GITHUB_COM_TOKEN=my-gh-token
```

And the following cli params:
`--platform github --onboarding true --automerge false --log-level debug`

I've also tried it with 
```
RENOVATE_TOKEN=my-ghe-token
RENOVATE_ENDPOINT=https://github.myoffice.com/api/v3/
GITHUB_COM_TOKEN=my-gh-token
``` 
as suggested here: https://github.com/renovatebot/renovate/issues/1685

And also by adding the following the the cli command:
```
--token my-token --endpoint https://github.myoffice.com/api/v3/
```
As suggested here: https://github.com/renovatebot/renovate/issues/3026

**Did you see anything helpful in debug logs?**
The following repeats many times:
```
INFO: getReleaseList error (repository=someowner/somepackage, dependencies=@myoffice/package-a,@myoffice/package-b,@myoffice/package-c, branch=renovate/minorpatch-@myofficepackage)
       "err": {
         "name": "GitHubError",
         "hostname": "github.myoffice.com",
         "method": "GET",
         "path": "/repos/someowner/somepackage/releases?per_page=100",
         "protocol": "https:",
         "url": "https://github.myoffice.com/repos/someowner/somepackage/releases?per_page=100",
         "gotOptions": {
           "path": "/repos/someowner/somepackage/releases?per_page=100",
           "protocol": "https:",
           "hostname": "github.myoffice.com",
           "hash": "",
           "search": "?per_page=100",
           "pathname": "/repos/someowner/somepackage/releases",
           "href": "https://github.myoffice.com/repos/someowner/somepackage/releases?per_page=100",
           "retry": {"methods": {}, "statusCodes": {}, "errorCodes": {}},
           "headers": {
             "user-agent": "https://github.com/sindresorhus/gh-got",
             "accept": "application/vnd.github.v3+json",
             "accept-encoding": "gzip, deflate",
             "authorization": "token my-ghe-token"
           },
           "hooks": {
             "beforeRequest": [],
             "beforeRedirect": [],
             "beforeRetry": [],
             "afterResponse": []
           },
           "decompress": true,
           "throwHttpErrors": true,
           "followRedirect": true,
           "stream": false,
           "form": false,
           "json": true,
           "cache": false,
           "useElectronNet": false,
           "baseUrl": "https://github.myoffice.com/api/v3/",
           "name": "GitHub",
           "platform": "github",
           "token": "my-ghe-token",
           "default": true,
           "method": "GET"
         },
         "statusCode": 406,
         "statusMessage": "Not Acceptable",
         "headers": {
           "server": "GitHub.com",
           "date": "Wed, 16 Jan 2019 16:46:11 GMT",
           "content-type": "text/html; charset=utf-8",
           "transfer-encoding": "chunked",
           "connection": "close",
           "status": "406 Not Acceptable",
           "cache-control": "no-cache, no-store",
           "vary": "X-PJAX",
           "x-ua-compatible": "IE=Edge,chrome=1",
           "set-cookie": [
             "_fi_sess=somerealcookie; path=/; secure; HttpOnly"
           ],
           "x-request-id": "51161d42a816c1f5ee243dccce7f7bc5",
           "x-runtime": "0.007010",
           "x-github-request-id": "7cb2c21a-fc01-4061-b865-44ac21c14ae6",
           "content-security-policy": "default-src 'none'; base-uri 'self'; block-all-mixed-content; child-src 'self'; connect-src 'self' github.myoffice.com status.github.com wss://github.myoffice.com; font-src 'self'; form-action 'self' github.myoffice.com; frame-ancestors 'none'; img-src * data:; manifest-src 'self'; media-src 'none'; script-src 'self'; style-src 'unsafe-inline' 'self'; worker-src 'self'",
           "strict-transport-security": "max-age=31536000; includeSubdomains",
           "x-content-type-options": "nosniff",
           "x-frame-options": "deny",
           "x-xss-protection": "1; mode=block",
           "x-runtime-rack": "0.012435"
         },
         "body": " ",
         "message": "undefined (406)",
         "stack": "GitHubError: undefined (406)\n    at EventEmitter.emitter.on (/Users/gordlea/dev/renovate-bot/node_modules/got/source/as-promise.js:74:19)\n    at <anonymous>\n    at process._tickCallback (internal/process/next_tick.js:189:7)"
       }
```

**To Reproduce**

1.  Create repository with '...'
2.  Configure '....'

**Expected behavior**
It should use the correct url:
`https://github.mycompany.com/api/v3/repos/someowner/somepackage/releases?per_page=100`

### PR DESCRIPTION
**Pull request by gordlea**
The code to get a list of releases was not using the RENOVATE_ENDPOINT when the npm modules it was fetching releases for had it's git repository on github enterprise.

Closes #3079

### CODE REVIEW

**Comment by gordlea** [comment]
Closing until i fix the tests

File: lib/workers/pr/changelog/source-github.js [hunk 1]
```diff
@@ -60,9 +60,10 @@ async function getChangeLogJSON({
     host: host === 'github.com' ? 'api.github.com' : host,
   });
   if (!config) {
-    logger.debug('Repository URL does not match any hnown hosts');
+    logger.debug('Repository URL does not match any known hosts');
     return null;
   }
+  const githubApiBaseURL = config.endpoint;
```
Comment (rarkins) [inline lib/workers/pr/changelog/source-github.js]: Does this still work for GHE when you need to fetch release notes from github.com? I may be misunderstanding it but it seems to be hardcoding the API lookup to be off GHE when most packages would be looked up from github.com still.

**Comment by gordlea** [comment]
Hi @rarkins, of the cases you list, I’ve definitely tested 2 and 4.

Just on my phone now, but I will post a more in depth explanation when I have some time this weekend. I will also try out the other test cases.

File: lib/workers/pr/changelog/source-github.js (hunk 1 above)
Comment (gordlea) [reply lib/workers/pr/changelog/source-github.js]: When debugging, this appears to be set to `'https://api.github.com` when the release notes are github.com, which is great. Originally I had done it a different way where I was comparing the url to see if it was from the same repository as the `RENOVATE_ENDPOINT`, etc, but this made the change much simpler.

Comment (rarkins) [reply lib/workers/pr/changelog/source-github.js]: I need to dig into why that is and also try to reduce the chance we don't break it in future :)

**Comment by gordlea** [comment]
It's funny, I verified that it works for a GHE repo with release notes on github.com, but now I can't seem to get it to work for GHE enterprise release notes for the test repo I created on our github enterprise. It also seems that GHE repo with GHE notes don't seem to work with the latest released renovate either (with no errors showing), which makes me think it might be a setup thing on my end?

All I need for release notes to be read from a dependency is a repo with a git tag and github release with some notes in it right?

I'm almost certain it was working on friday. I'll go into it in more detail on monday and try to verify.

**Comment by rarkins** [comment]
As far as I’m aware, the released version of Renovate doesn’t fetch release notes from GHE either

**Comment by rarkins** [comment]
Assuming it’s an npm package:
- the package must point to the GHE repo as its source  in its npm metadata
- the source must have a GitHub release matching the version, or a changelog file with a heading matching it. Much easier to test with the releases though

**Comment by rarkins** [comment]
And don’t forget the bot/token needs access to that source repo too, not just the downstream one it’s renovating at the time

**Comment by gordlea** [comment]
Ah it was access was missing. So I can confirm that it will create release notes from deps hosted on the same GHE as the package being renovated, which is great. I was pretty sure that worked, because, but you made me doubt myself on that one, glad I'm not going crazy. 😄  

I've also tried all of the 4 scenarios you named, and they all appear to work fine.

**Comment by rarkins** [comment]
@gordlea thanks for that confirmation. I'll do some testing myself and then merge

**Pull request merged**